### DISCUSSION TITLE
Plugin development (comment overlay)

### DISCUSSION THREAD

**Discussion by VINT64** [d2phap/ImageGlass#449] [General] [answered]
Good time of day,

I have been looking for an image viewer with possibility to add overlay over an image. 
Having found none, I would like someone to write a plugin for some free image viewer, or to write such plugin on my own. 

This kind of overlay is used on anime images' sites like -booru to add translations over the image. When you open an image on the site, it has semi-transparent rectangles over an image. If you hover over a rectangle, a comment will pop-up. ![example](https://user-images.githubusercontent.com/33546510/48003064-a5f63280-e10d-11e8-8868-632f5e823e87.png)
By single click you can hide/show all these rectangles. Very comfortable to view image.

I would like to use this plugin to provide manga translations to readers. Normally, each user has to buy digital version of manga on his own, but nowadays translators have to edit pages in graphics editor like Paint.NET or Photoshop, and provide 'as is'.
I have an idea to keep translation (or comments, in general) in structured JSON file, and provide a pack of such files to readers, letting them to find the images on their own.

Of course I should provide readers with a program to open these images along with comment files, and instruction, how to install and configure such a program.

I would like to know your opinion on all this, whether this image viewer is suitable for this goal, and any advices which can help me. 

Thanks for reading,

**Comment by VINT64** [comment]
[In my fork](https://github.com/VINT64/ImageGlass/commit/f8bdbcc5f1139bd66bb376c7a74202bba7086cfe#diff-1c9c933f3ad3aa1ccb0f2d97e360f712) I tried to fix your old demo plugin to test it and learn, how plugins work. 
I guess I succeeded to compile it in .dll file, but now I'm completely lost and don't know, what to do next to connect it to a local installed app and to test it.
As I almost don't know Visual Studio and C#, I ask for your help to guide me through plugin creation.

**Comment by d2phap** [comment, marked as answer]
Hi @VINT64 

> I would like to know your opinion on all this, whether this image viewer is suitable for this goal, and any advice which can help me.

Unfortunately, the current application doesn't support plugins. 
ImageGlass goals are simple and lightweight. Having too many features makes it fat and heavy. The new features should be for the "viewing photos" purpose.

Your suggestion is more suitable for comic reader than a pure photo viewer.
If you want to develop this hover feature, you can fork the repository and feel free to change the code (in the project _ImageGlass.ImageBox_), and build your new version.

**Comment by fire-eggs** [comment]
> If you want to develop this hover feature, you can fork the repository and feel free to change the code

You'll want to work with a code snapshot prior to April 21 2018, which is when the plugin support was removed.
